components/vfs/src/std_filesystem.cpp

```diff
@@ -0,0 +1,292 @@
+/* Copyright (c) 2022, Arm Limited and Contributors
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 the "License";
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "components/vfs/std_filesystem.hpp"
+
+#include <cassert>
+#include <filesystem>
+#include <fstream>
+#include <iostream>
+#include <limits>
+#include <sstream>
+
+#include "components/vfs/helpers.hpp"
+
+namespace components
+{
+namespace vfs
+{
+
+inline std::filesystem::path get_full_path(const std::filesystem::path &base, const std::string &path)
+{
+	auto parts = helpers::get_individual_directory_parts(path);
+
+	auto full_path = base;
+
+	for (auto &part : parts)
+	{
+		full_path /= part;
+	}
+
+	full_path /= helpers::get_file_name(path);
+
+	return full_path;
+}
+
+StdFSFileSystem::StdFSFileSystem(const std::filesystem::path &base_path) :
+    FileSystem{},
+    m_base_path{base_path}
+{
+}
+
+StdFSTempFileSystem::StdFSTempFileSystem() :
+    StdFSFileSystem{}
+{
+	std::error_code error;
+
+	m_base_path = std::filesystem::temp_directory_path(error);
+
+	if (error)
+	{
+		throw std::runtime_error{"failed to initialize temporary file directory path"};
+	}
+}
+
+bool StdFSFileSystem::folder_exists(const std::string &file_path)
+{
+	return std::filesystem::is_directory(m_base_path += file_path);
+}
+
+bool StdFSFileSystem::file_exists(const std::string &file_path)
+{
+	return std::filesystem::is_regular_file(m_base_path += file_path);
+}
+
+StackErrorPtr StdFSFileSystem::read_file(const std::string &file_path, std::shared_ptr<Blob> *blob)
+{
+	if (!blob)
+	{
+		return nullptr;
+	}
+
+	auto full_path = get_full_path(m_base_path, file_path);
+
+	if (!std::filesystem::exists(full_path))
+	{
+		return StackError::unique("file does not exist: " + full_path.string(), "vfs/std_filesystem.cpp", __LINE__);
```

Comment:
`__FILE__` is an absolute path. In some cases this can be hundreds of characters long. It is also not consistent across platforms so there is no easy way to shorten this. `"vfs/std_filesystem.cpp"` this style is not correct in terms of filesystem path but will still be easy to read from a log output.